server/admin/admin_server_test.go

```diff
@@ -65,6 +66,23 @@ func TestServer_BeginError(t *testing.T) {
 	ctrl := gomock.NewController(t)
 	defer ctrl.Finish()
 
+	// PEM on the testonly trees is ECDSA, so let's use an ECDSA key for tests.
+	privateKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		t.Fatalf("Error generating test key: %v", err)
+	}
+
+	publicKeyDER, err := x509.MarshalPKIXPublicKey(privateKey.Public())
+	if err != nil {
+		t.Fatalf("Error marshaling public key: %v", err)
+	}
+
+	// Need to change the public key to correspond with the private key generated above.
+	validTree := *testonly.LogTree
+	validTree.PublicKey = &trillian.PublicKey{
+		Der: publicKeyDER,
```

Comment:
It will do if `PublicKey` is nil, but it's not nil in `testonly.LogTree`. This test could just do `validTree.PublicKey = nil` though, for simplicity. Would you like me to make that change?